modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/VirtualFlowTest.java

```diff
@@ -1126,6 +1126,44 @@ public void test_RT39568() {
         assertMinimalNumberOfCellsAreUsed(flow);
         assertEquals(flow.getViewportLength()-25.0, VirtualFlowShim.<IndexedCell>cells_getLast(flow.cells).getLayoutY(), 0.0);
     }
+
+    @Test
+    // see JDK-8197536
+    public void testScrollOneCell() {
+        flow.resize(400, 400);
+
+        int total = 10000;
+        flow.setCellCount(total);
+        pulse();
+
+        int count = 9000;
+        flow.setPosition(0d);
+        pulse();
+        flow.setPosition(((double)count) / total);
+        pulse();
+
+        //simulate 500 right key strokes
+        for (int i = 0; i < 500; i++) {
+            count++;
+            flow.scrollTo(count);
+            pulse();
+        }
+
+        IndexedCell vc = flow.getCell(count);
+
+        double cellPosition = flow.getCellPosition(vc);
+        double cellLength = flow.getCellLength(count);
+        double viewportLength = flow.getViewportLength();
+
+        assertEquals("Last cell must end on viewport size", viewportLength, (cellPosition + cellLength), 0.1);
+    }
+
+    @Test
+    // see JDK-8197536
+    public void testScrollOneCellHorizontal() {
+        flow.setVertical(false);
+        testScrollOneCell();
```

Comment:
Minor: Normally we wouldn't call one test method from another. Rather we typically create a utility method that both test methods would call.

Reply:
Moved to a method as requested. Thanks.